tests/test_models.py

```diff
@@ -1,341 +1,122 @@
         dm.dq = data
 
 
-@pytest.mark.skip("requires ImageModel")
 def test_broadcast2():
-    with ImageModel() as dm:
+    with BasicModel() as dm:
         data = np.empty((52, 50))
         dm.data = data
 
         dq = np.empty((50,))
         dm.dq = dq
 
 
-@pytest.mark.skip("requires jwst model implementations")
-def test_from_hdulist():
-    from astropy.io import fits
-    warnings.simplefilter("ignore")
-    with fits.open(FITS_FILE, memmap=False) as hdulist:
-        with datamodels.open(hdulist) as dm:
-            dm.data
-        assert not hdulist.fileinfo(0)['file'].closed
-
-
 def delete_array():
-    with datamodels.open() as dm:
-        del dm.data
-
-
-@pytest.mark.skip("base model no longer includes attribute meta.subarray")
-def test_subarray():
-    with DataModel(FITS_FILE) as dm:
-        dm.meta.subarray.xstart
-
+    with BasicModel() as m:
+        del m.data
 
-def roundtrip(func):
-    def _create_source():
-        dm = DataModel(FITS_FILE)
 
-        assert dm.meta.instrument.name == 'MIRI'
-
-        dm.meta.instrument.name = 'NIRCAM'
-        dm.meta.subarray.xstart = 42
-        return dm
-
-    def _check_output(dm):
-        assert dm.meta.instrument.name == 'NIRCAM'
-        assert dm.meta.subarray.xstart == 42
-
-    def test():
-        with _create_source() as dm:
-            with func(dm) as dm2:
-                _check_output(dm2)
-
-    test.__name__ = func.__name__
-
-    return test
-
-
-@pytest.mark.skip("base model no longer includes attribute meta.instrument")
-@roundtrip
-def test_from_fits_write(dm):
-    dm.to_fits(TMP_FITS, overwrite=True)
-    return DataModel.from_fits(TMP_FITS)
-
-
-@pytest.mark.skip("base model no longer includes attribute meta.instrument")
 def test_delete():
-    with DataModel() as dm:
-        dm.meta.instrument.name = 'NIRCAM'
-        assert dm.meta.instrument.name == 'NIRCAM'
-        del dm.meta.instrument.name
-        assert dm.meta.instrument.name is None
-
+    with BasicModel() as dm:
+        dm.meta.telescope= 'JWST'
+        assert dm.meta.telescope == 'JWST'
+        del dm.meta.telescope
+        assert dm.meta.telescope is None
 
-@pytest.mark.skip("requires QuadModel")
-def test_open():
-    with datamodels.open():
-        pass
 
-    with datamodels.open((50, 50)):
-        pass
-
-    with pytest.warns(stdatamodels.util.NoTypeWarning):
-        with datamodels.open(FITS_FILE) as dm:
-            assert isinstance(dm, QuadModel)
-
-
-@pytest.mark.skip("requires jwst model implementations")
-def test_open_warning():
-    with warnings.catch_warnings(record=True) as warners:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
-        with datamodels.open(FITS_FILE) as model:
-            pass
-
-        class_name = model.__class__.__name__
-        j = None
-        for i, w in enumerate(warners):
-            if class_name in str(w.message):
-                j = i
-        assert j is not None
```

Comment:
These 2 tests are deleted since `open` is no longer implemented.  We should probably keep these over in `jwst.datamodels`, and clean up that second one to use `with pytest.warns()`.

Reply:
Yes there will have to be a wave of cleanup over in jwst once the tests have been sorted between the two packages.

Reply:
🥽